kuma/users/tests/test_templates.py

```diff
@@ -1296,30 +1255,98 @@ def test_delete_link_does_not_appear_summary_page_other_revisions(self):
             'wiki.delete_document',
             kwargs={'document_path': doc1.slug},
             force_locale=True)
-        # TODO: PhaseIII
-        # delete_url_reverted = reverse(
-        #     'wiki.delete_document',
-        #     kwargs={'document_path': doc2.slug},
-        #     force_locale=True)
+        delete_url_reverted = reverse(
+            'wiki.delete_document',
+            kwargs={'document_path': doc2.slug},
+            force_locale=True)
         # TODO: PhaseIV
         # delete_url_new_action
 
-        # TODO: PhaseIII
-        # delete_link_reverted_section = page.find('#reverted a[href="{url}"]'.format(
-        #     url=delete_url_reverted))
+        delete_link_reverted_section = page.find('#reverted a[href="{url}"]'.format(
+            url=delete_url_reverted))
         # TODO: PhaseIV
         # delete_link_new_action_section = page.find('#new-actions-by-user a[href="{url}"]'.format(
         #     url=delete_url_new_action))
         delete_link_already_spam_section = page.find('#already-spam a[href="{url}"]'.format(
             url=delete_url_already_spam))
 
         # There should not be a delete link in any of these sections
-        # TODO: PhaseIII
-        # eq_(len(delete_link_reverted_section), 0)
+        eq_(len(delete_link_reverted_section), 0)
         # TODO: PhaseIV
         # eq_(len(delete_link_new_action_section), 0)
         eq_(len(delete_link_already_spam_section), 0)
 
+    def test_newest_revision_is_not_spam(self):
+        """
+        Test with a spam user who has made revisions to a single document,
+        but another user has made a more recent revision.
+        The newest revision was created by a non-spam user,
+        so none of the revisions actually need to be reverted.
+        """
```

Comment:
I agree that if there are later commits, then automatic reverting is not possible. However, it still may need follow up by a moderator, maybe appearing in the "needs followup" section. Maybe. I'd need some practice with real spam to see how it works. No change for now.